enhancements/cluster-logging/fine-grained-per-namespace-logs-access.md

```diff
@@ -0,0 +1,249 @@
+
+### Non-Goals
+
+* Prevent users with elevated privileges from escalating their privileges to access application logs that they are not supposed to have access to. 
+
+## Proposal
+
+Use the `ClusterLogging` custom resource definition managed by the Cluster Logging Operator to enable/disable the fine grained access to logs for the LokiStack logstore. This will be achieved by adding a new field to the CRD called `advancedLogsAccess`. This new field is a boolean with a default value of false.
+
+* For the first use case, where we have non-admin users, the proposed solution is to create a group for application log readers called `log-reader-group`, and grant this group the ability to read application logs by binding it to the existing ClusterRole `logging-application-logs-reader`. The cluster admin can then either add a user to the `log-reader-group` thus granting access to application logs on all namespace where they have access, or create a role binding to the user on each namespace they want to grant access to logs on.
+
+* For the second use case, where we have users with admin like privileges, the proposed solution is to create a new group called `restricted-cluster-admin-group`. This group has all the permissions a cluster admin has, except for application logs. We can then grant this group access to application logs on specific namespace by binding it to `logging-application-logs-reader` on the desired namespace. 
+
+### Workflow Description
+
+1. The cluster administrator enables fine grained logs access in the `ClusterLogging` resource:
+
+```yaml
+apiVersion: logging.openshift.io/v1
+kind: ClusterLogging
+metadata:
+  name: instance
+  namespace: openshift-logging
+spec:
+  logStore:
+    type: lokistack
+    lokistack:
+      name: lokistack-dev
+      advancedLogsAccess: true
+  ...
+```
+
+2. The Cluster Logging Operator (CLO) deletes the `logging-all-authenticated-application-logs-reader` ClusterRoleBinding. This binding allows all authenticated users to see application logs in namespace where they have access.
+
+3. The CLO then creates a new empty group `log-reader-group` and binds it to `logging-application-logs-reader`:
+
+```yaml
+apiVersion: user.openshift.io/v1
+kind: Group
+metadata:
+  name: log-reader-group
+users:
+```
+The group contains no users. The cluster admin can then add users to the group by editing it manually.
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: logging-application-logs-reader-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: logging-application-logs-reader
+subjects:
+- kind: Group
+  name: log-reader-group
+  apiGroup: rbac.authorization.k8s.io
+```
+
+4. The operator also creates a new ClusterRole called `restricted-cluster-admin` and binds it to a new group `restricted-cluster-admin-group`. `restricted-cluster-admin` grants access to all api objects except for logs:
```

Comment:
This step to me feels weird... it looks like we are encouraging people to use `restricted-cluster-admin` whilst in reality, IIUC we want to give them the tools to be able to restrict log access to pre-defined (by them) ClusterRoles. This to me would be done through good documentation on how to configure a privileged ClusterRole and how to grant it log access (by creating the RoleBinding `logging-application-logs-reader`). Please correct me if I'm miss understanding something

Reply:
The ClusterRole created here and its binding are more for convenience. We can go without them, but I think for the case where there are cluster admins with restricted access to logs, creating this would make it easier to configure out of the box.